File: Java/Assignment7_1.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// file name   : Assignment7_1.java
// Description : write a program which will accepte the number from user and display below pattern.
// input       : 3
// Pattern.    : *   *   *  #   #   #
// input       : 5
// Pattern.    : *   *   *  *   *   #   #   #   #   #
// Author      : Aniket Devaba Markad                        
// Date        : 1/1/2026
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Pattern
{

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name : Display
// Description   : This function is used to print the pattern . 
// Author        : Aniket Devaba Markad 
// Date          : 1/1/2026 
// 
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    void Display(int iNo)
    {
        if(iNo < 0 )
        {
            iNo = -iNo;
        }

        int i = 0,iCnt = 0;
        for( i = 1; i <= iNo; i++)
        {
            System.out.print(" * \t");
            iCnt++;
        }

        for(i = 1;i <= iCnt ; i++)
        {
            System.out.print("# \t");
        }
    }
}
public class Assignment7_1
{
    public static void main(String a[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("enter the number ");
        int iValue = sobj.nextInt();

        Pattern pobj = new Pattern();
        pobj.Display(iValue);

    }
}

File: CPP/Assignment7_1.cpp
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//  file name   : Assignment7_1.cpp 
//  Description : Write a program which accept number from user and print pattern n = * & #
//  Author      : Aniket Devaba Markad 
//  Date        : 10/12/2025
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include <iostream>
using namespace std;

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//  Function Name : DisplayPattern
//  Description   : This function is used to print pattern: first n '*' then n '#' 
//  Input         : int 
//  Output        : void
//  Author        : Aniket Devaba Markad 
//  Date          : 10/12/2025
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

void DisplayPattern(int iNo)
{
    if(iNo < 0)
        iNo = -iNo;

    for(int iCnt = 1; iCnt <= 2 * iNo; iCnt++)
    {
        if(iCnt <= iNo)
            cout << "*\t";
        else
            cout << "#\t";
    }
    cout << endl;
}

int main()
{
    int iValue = 0;

    cout << "Enter number: ";
    cin >> iValue;

    DisplayPattern(iValue);

    return 0;
}

// Time Complexity: O(2n)
